src/NetMQ/Core/YPipe.cs

```diff
@@ -179,7 +178,7 @@ public bool CheckRead()
         /// Attempts to read an item from the pipe.
         /// </summary>
         /// <returns><c>true</c> if the read succeeded, otherwise <c>false</c>.</returns>
-        public bool TryRead(out T value)
+        public bool TryRead([MaybeNull] out T value)
```

Comment:
NotNullWhen?

Reply:
You're right. Updated.

Actually in this case it must be `MaybeNullWhen` because when `T` is unconstrained, it might be a value type for which `T?` actually means `Nullable<T>`. In C# 8 you cannot annotate an unconstrained generic type with `?`.

In C# 9 the team is investigating allowing `T??` as a syntax for exactly this situation.